azurerm/resource_arm_stream_analytics_reference_input_blob.go

```diff
@@ -0,0 +1,243 @@
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"stream_analytics_job_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"date_format": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"path_pattern": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"storage_account_key": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"storage_account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"storage_container_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"time_format": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"serialization": azure.SchemaStreamAnalyticsStreamInputSerialization(),
+		},
+	}
+}
+
+func resourceArmStreamAnalyticsReferenceInputBlobCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).streamAnalyticsInputsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure Stream Analytics Reference Input Blob creation.")
+	name := d.Get("name").(string)
+	jobName := d.Get("stream_analytics_job_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, jobName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Stream Analytics Stream Input %q (Job %q / Resource Group %q): %s", name, jobName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_stream_analytics_stream_input_blob", *existing.ID)
+		}
+	}
+
+	containerName := d.Get("storage_container_name").(string)
+	dateFormat := d.Get("date_format").(string)
+	pathPattern := d.Get("path_pattern").(string)
+	storageAccountKey := d.Get("storage_account_key").(string)
+	storageAccountName := d.Get("storage_account_name").(string)
+	timeFormat := d.Get("time_format").(string)
+
+	serializationRaw := d.Get("serialization").([]interface{})
+	serialization, err := azure.ExpandStreamAnalyticsStreamInputSerialization(serializationRaw)
+	if err != nil {
+		return fmt.Errorf("Error expanding `serialization`: %+v", err)
+	}
+
+	props := streamanalytics.Input{
+		Name: utils.String(name),
+		Properties: &streamanalytics.ReferenceInputProperties{
+			Type: streamanalytics.TypeReference,
```

Comment:
This value is what sets it distinguishes a reference input from a streaming input to the stream analytics job.